scripts/zones/VeLugannon_Palace/mobs/Zipacna.lua

```diff
@@ -4,122 +4,523 @@
-    mob:pathThrough(pathNodes, xi.path.flag.PATROL)
+
+    -- Zip popped on blue short side (!pos -196, 0, 396)
+    if pos.x <= -195 then
+        mob:pathThrough(blueShortSideSpawnPath, xi.path.flag.COORDS)
+
+    -- Zip popped at the blue basement spawn point (!pos -121, 16, 420)
+    elseif pos.x <= -120 then
+        mob:pathThrough(blueBasementSpawnPath, xi.path.flag.COORDS)
+
+    -- Zip popped at the Yellow Short Side Spawn Point - going to first yellow door (!pos 191, 0, 399)
+    elseif pos.x >= 190 then
+        mob:pathThrough(yellowShortSideSpawnPath, xi.path.flag.COORDS)
+
+    -- Zip popped in the yellow side basement - going to first yellow door
+    elseif pos.x >= 140 then
+        mob:pathThrough(yellowBasementSpawnPath, xi.path.flag.COORDS)
+    end
+end
+
+entity.onMobFight = function(mob, target)
 end
 
 entity.onMobDeath = function(mob, player, optParams)
 end
 
+entity.onMobRoam = function(mob)
+    local pos              = mob:getPos()
+    local firstBlueDoor    = GetNPCByID(ID.npc.B_DOOR_OFFSET + 1)
+    local middleBlueDoor   = GetNPCByID(ID.npc.B_DOOR_OFFSET)
+    local firstYellowDoor  = GetNPCByID(ID.npc.Y_DOOR_OFFSET + 1)
+    local middleYellowDoor = GetNPCByID(ID.npc.Y_DOOR_OFFSET)
+    local getLastDoor      = mob:getLocalVar('[Zipacna]lastDoor')
+
+    -- begin of door checks. System in place is to check distance betwen mob and the NPC door (mapped to IDs.lua in local firstBlueDoorCoord, etc)
+    if
+        mob:checkDistance(firstBlueDoorCoord.x, firstBlueDoorCoord.y, firstBlueDoorCoord.z) <= 5
+    then
+        if -- POS from path to first blue door if Zip is coming from the basement
+            pos.x <= -162.000 and
+            pos.y <= 1 and
+            pos.z >= 379 and
+            getLastDoor ~= 1
+        then
+            if
+                firstBlueDoor and
+                firstBlueDoor:getAnimation() == xi.anim.CLOSE_DOOR
+            then
+                mob:pathThrough(firstBlueToBasement, xi.path.flag.COORDS)
+            elseif
+                firstBlueDoor and
+                firstBlueDoor:getAnimation() == xi.anim.OPEN_DOOR
+            then
+                mob:pathThrough(firstBluetoBlueMiddle, xi.path.flag.COORDS)
+                mob:setLocalVar('[Zipacna]lastDoor', 1)
+            end
+        elseif -- coming from middle blue door to first blue door
+            pos.x >= -158.000 and
+            pos.y <= 1 and
+            pos.z >= 379 and
+            getLastDoor >= 1
+        then
+            if
+                firstBlueDoor and
+                firstBlueDoor:getAnimation() == xi.anim.CLOSE_DOOR
+            then
+                mob:pathThrough(firstBluetoBlueMiddle, xi.path.flag.COORDS)
+            elseif
+                firstBlueDoor and
+                firstBlueDoor:getAnimation() == xi.anim.OPEN_DOOR
+            then
+                mob:pathThrough(firstBlueToBasement, xi.path.flag.COORDS)
+            end
+        end
+    end
+
+    -- middle blue door
+    if
```

Comment:
You repeat this exact pattern 4 times using different points and paths, extract it out into a function or a table, etc.

Reply:
Helping Hook out with this - So will comment on some of this on his behalf.
Extracted these repeated calls into a helper function that determines the path finding logic.